### PR TITLE
Buildings v2

### PR DESCRIPTION
**Pull request by DavidPgl**

### CODE REVIEW

File: Assets/Scripts/Buildings/Building.cs [hunk 1]
```diff
@@ -4,66 +4,70 @@
...
+        /// Whether the building is build or not. Invokes <see cref="OnBuild"/>
         /// </summary>
-        public bool IsBuild { get; set; }
+        public bool IsBuild
+        {
+            get => _isBuild;
+            set
+            {
+                _isBuild = value;
+                OnBuild();
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Buildings/Building.cs]: Could you add a check to see if the property is actually set to `true` before executing `OnBuild()`?

File: Assets/Scripts/Buildings/Campsite.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        /// <summary>
+        /// Whether this campsite is already unlocked.
+        /// </summary>
+        public bool IsUnlocked
+        {
+            get => isUnlocked;
+            set
+            {
+                isUnlocked = value;
+                OnUnlock?.Invoke();
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Buildings/Campsite.cs]: Please also add a check here (see above)

File: Assets/Scripts/Managers/PlayerManager.cs [hunk 3]
```diff
@@ -5,7 +5,11 @@ namespace Managers
     public class PlayerManager : Singleton<PlayerManager>
     {
         [SerializeField] private GameObject player;
-        
+
         public GameObject GetPlayer() => player;
+
+        public bool PlayerInReach(GameObject otherObject, float distanceToPlayer) =>
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Managers/PlayerManager.cs]: Maybe rename this to something like `reachDistance` or `maxRange` to make it clearer what's actually meant? And/Or add documentation :P

**Review by Kaes3kuch3n — DISMISSED** [review]

File: Assets/Scripts/Buildings/Building.cs (hunk 1 above)
Comment (DavidPgl) [reply Assets/Scripts/Buildings/Building.cs]: Hahaha, that's how it should work 😂 Will do

**Review by SamuelllG — APPROVED** [review]
👌

**Review by inkompotato — APPROVED** [review]

**Pull request merged**